code/modules/hivemind/machines.dm

```diff
@@ -0,0 +1,486 @@
+		if(target)
+			if(get_dist(src, target) <= 1)
+				icon_state = "core_new-fear"
+			else
+				icon_state = "core_new-see"
+				dir = get_dir(src, target)
+		else
+			icon_state = initial(icon_state)
+
+
+/obj/structure/hivemind_machine/node/update_icon()
+	overlays.Cut()
+	if(stunned)
+		icon_state = "core_new-disabled"
+		overlays += "core-smirk_disabled"
+	else
+		icon_state = initial(icon_state)
+		overlays += "core-smirk"
+
+
+/obj/structure/hivemind_machine/node/use_ability(atom/target)
+	hive_mind_ai.get_points()
+
+
+/obj/structure/hivemind_machine/node/name_pick()
+	name = "[hive_mind_ai.name] [hive_mind_ai.surname]" + " [rand(999)]"
+
+
+//there we binding or un-binding hive with wire
+//in this way, when our node will be destroyed, wireweeds will die too
+/obj/structure/hivemind_machine/node/proc/add_wireweed(obj/effect/plant/hivemind/wireweed)
+	if(wireweed.master_node)
+		wireweed.master_node.remove_wireweed(wireweed)
+	wireweed.master_node = src
+	my_wireweeds.Add(wireweed)
+
+/obj/structure/hivemind_machine/node/proc/remove_wireweed(obj/effect/plant/hivemind/wireweed)
+	my_wireweeds.Remove(wireweed)
+	wireweed.master_node = null
+
+//there we check for other nodes
+//if no any other hives will be found, game over
+/obj/structure/hivemind_machine/node/proc/check_for_other()
+	if(hive_mind_ai)
+		if(!hive_mind_ai.hives.len)
+			hive_mind_ai.die()
+
+
+//TURRET
+//shooting the target with toxic goo
+/obj/structure/hivemind_machine/turret
+	name = "shooter"
+	desc = "Strange thing with some kind of tube."
+	max_health = 140
+	icon_state = "turret"
+	cooldown_time = 5 SECONDS
+	var/proj_type = /obj/item/projectile/goo
+
+
+/obj/structure/hivemind_machine/turret/activate()
+	for(var/mob/living/target in view(src))
+		if(target.stat == CONSCIOUS && target.faction != "hive")
+			use_ability(target)
+			set_cooldown()
+			break
+
+
+/obj/structure/hivemind_machine/turret/use_ability(atom/target)
+	var/obj/item/projectile/proj = new proj_type(loc)
+	proj.launch(target)
+	playsound(src, 'sound/effects/blobattack.ogg', 70, 1)
+
+
+
+//MOB PRODUCER
+//spawns mobs from list
+/obj/structure/hivemind_machine/mob_spawner
+	name = "assembler"
+	desc = "Cylindrical machine with some lights and entry port. You can hear how something moves inside."
+	max_health = 120
+	icon_state = "spawner"
+	cooldown_time = 10 SECONDS
+	var/mob_to_spawn
+	var/mob_amount = 1
+
+/obj/structure/hivemind_machine/mob_spawner/Initialize()
+	..()
+	mob_to_spawn = pick(/mob/living/simple_animal/hostile/hivemind/stinger, /mob/living/simple_animal/hostile/hivemind/bomber)
+
+
+/obj/structure/hivemind_machine/mob_spawner/activate()
+	if(!mob_to_spawn || spawned_creatures.len >= mob_amount)
+		return
+	if(locate(/mob/living) in loc)
+		return
+
+	//here we upgrading our spawner and rise controled mob amount, based on EP
+	if(hive_mind_ai.evo_points > 100)
+		mob_amount = 2
+	else if(hive_mind_ai.evo_points > 300)
+		mob_amount = 3
+
+	for(var/mob/living/target in view(src))
```

Comment:
This won't find people in mechas.
It may be better to redesign this as a turret, since they have better code for detecting people in vehicles. Or at least look at turret code, i believe what you want is mobs_in_view